timeseries/src/autogluon/timeseries/models/presets.py

```diff
@@ -18,109 +19,154 @@
-            "SeasonalNaive": {},
-            "AutoETS": {},
-            "NPTS": {},
-            "DynamicOptimizedTheta": {},
-            "RecursiveTabular": {},
-            "DirectTabular": {},
-            "TemporalFusionTransformer": {},
-            "PatchTST": {},
-            "DeepAR": {},
-            "Chronos": [
-                {
-                    "ag_args": {"name_suffix": "ZeroShot"},
-                    "model_path": "bolt_base",
-                },
-                {
-                    "ag_args": {"name_suffix": "FineTuned"},
-                    "model_path": "bolt_small",
-                    "fine_tune": True,
-                    "target_scaler": "standard",
-                    "covariate_regressor": {"model_name": "CAT", "model_hyperparameters": {"iterations": 1_000}},
-                },
-            ],
-            "TiDE": {
-                "encoder_hidden_dim": 256,
-                "decoder_hidden_dim": 256,
-                "temporal_hidden_dim": 64,
-                "num_batches_per_epoch": 100,
-                "lr": 1e-4,
-            },
-        },
+class TrainableModelSetBuilder:
+    """Responsible for building a list of model objects, in priority order, that will be trained by the
+    Trainer."""
+
+    VALID_AG_ARGS_KEYS = {
+        "name",
+        "name_prefix",
+        "name_suffix",
     }
-    return default_model_hps[key]
-
-
-def get_preset_models(
-    freq: Optional[str],
-    prediction_length: int,
-    path: str,
-    eval_metric: Union[str, TimeSeriesScorer],
-    hyperparameters: Union[str, Dict, None],
-    hyperparameter_tune: bool,
-    covariate_metadata: CovariateMetadata,
-    all_assigned_names: List[str],
-    excluded_model_types: Optional[List[str]],
-    multi_window: bool = False,
-    **kwargs,
-):
-    """
-    Create a list of models according to hyperparameters. If hyperparamaters=None,
-    will create models according to presets.
-    """
-    models = []
-    hyperparameter_dict = get_hyperparameter_dict(hyperparameters, hyperparameter_tune)
 
-    model_priority_list = sorted(
-        hyperparameter_dict.keys(), key=lambda x: ModelRegistry.get_model_priority(x), reverse=True
-    )
-    excluded_models = get_excluded_models(excluded_model_types)
-    all_assigned_names = all_assigned_names.copy()
+    def __init__(
+        self,
+        path: str,
+        freq: Optional[str],
+        prediction_length: int,
+        eval_metric: TimeSeriesScorer,
+        target: str,
+        quantile_levels: List[float],
+        covariate_metadata: CovariateMetadata,
+        banned_model_names: List[str],
+        multi_window: bool,
+    ):
+        self.path = path
+        self.freq = freq
+        self.prediction_length = prediction_length
+        self.eval_metric = eval_metric
+        self.target = target
+        self.quantile_levels = quantile_levels
+        self.covariate_metadata = covariate_metadata
+        self.banned_model_names = banned_model_names.copy()
```

Comment:
If the purpose of `__init__` is to separate the task definition from the "actual inputs needed to build the model set", I would recommend making this class stateless by passing `banned_model_names` as an argument to `get_model_set`. This looks like a small price to pay to make the method pure and avoid ever needing to think about the state of the model builder object.